README.md

```diff
@@ -131,113 +113,151 @@ This will output:
+	"sync"
 )
 
-func main() {
-   usingGalacticChannels()
-}
-
-func usingGalacticChannels() {
-
-   // get a pointer to the bus.
-   b := bus.GetBus()
-
-   // get a pointer to the channel manager
-   cm := b.GetChannelManager()
-
-   channel := "my-stream"
-   cm.CreateChannel(channel)
-
-   // create done signal
-   var done = make(chan bool)
-
-   // listen to stream of messages coming in on channel.
-   h, err := b.ListenStream(channel)
-
-   if err != nil {
-      log.Panicf("unable to listen to channel stream, error: %e", err)
-   }
-
-   count := 0
-
-   // listen for ten messages and then exit, send a completed signal on channel.
-   h.Handle(
-      func(msg *model.Message) {
-
-         // unmarshal the payload into a Response object (used by fabric services)
-         r := &model.Response{}
-         d := msg.Payload.([]byte)
-         json.Unmarshal(d, &r)
-         fmt.Printf("Stream Ticked: %s\n", r.Payload.(string))
-         count++
-         if count >=10 {
-            done <- true
-         }
-      },
-      func(err error) {
-         log.Panicf("error received on channel %e", err)
-      })
-
-   // create a broker connector config, in this case, we will connect to the demo endpoint.
-   config := &bridge.BrokerConnectorConfig{
-      Username:   "guest",
-      Password:   "guest",
-      ServerAddr: "localhost:8090",
-      UseWS:      true,
-      WebSocketConfig: &bridge.WebSocketConfig{
-         WSPath:    "/fabric",
-      }}
-
-   // connect to broker.
-   c, err := b.ConnectBroker(config)
-   if err != nil {
-      log.Panicf("unable to connect to fabric, error: %e", err)
-   }
-
-   // mark our local channel as galactic and map it to our connection and the /topic/simple-stream service
-   // running on localhost:8090
-   err = cm.MarkChannelAsGalactic(channel, "/topic/simple-stream", c)
-   if err != nil {
-      log.Panicf("unable to map local channel to broker destination: %e", err)
-   }
-
-   // wait for done signal
-   <-done
-
-   // mark channel as local (unsubscribe from all mappings)
-   err = cm.MarkChannelAsLocal(channel)
-   if err != nil {
-      log.Panicf("unable to unsubscribe, error: %e", err)
-   }
-   err = c.Disconnect()
-   if err != nil {
-      log.Panicf("unable to disconnect, error: %e", err)
-   }
+// SimpleStream will connect to our demo broker running at transport-bus.io, listen to a simple stream that is being
+// broadcast on /topic/simple-stream. Every second a random word is broadcast on that channel to anyone listening.
+// This should take 10 seconds to run.
+func SimpleStream() []string {
+
+	// get a pointer to the bus.
+	b := bus.GetBus()
+
+	// get a pointer to the channel manager
+	cm := b.GetChannelManager()
+
+	channel := "my-stream"
+	cm.CreateChannel(channel)
+
+	// listen to stream of messages coming in on channel, a handler is returned that allows you to add in
+	// lambdas that handle your
+	handler, err := b.ListenStream(channel)
```

Comment:
just a suggestion, but maybe we can pull this code closer to where handler is actually defined, like L191 so the flow is smoother to readers?

Reply:
good idea. done